components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -67,7 +71,9 @@ public class ActionExecutorServiceImpl implements ActionExecutorService {
     private static final Log LOG = LogFactory.getLog(ActionExecutorServiceImpl.class);
 
     private static final ActionExecutorServiceImpl INSTANCE = new ActionExecutorServiceImpl();
+    private static final ActionExecutionDiagnosticLogger diagnosticLogger = new ActionExecutionDiagnosticLogger();
     private final APIClient apiClient;
+    private final ExecutorService executorService = ThreadLocalAwareExecutors.newFixedThreadPool(1);
```

Comment:
A thread pool with 1 thread will not help on the real intention to run this asynchronously.

Common recommendation seems to be " Runtime.getRuntime().availableProcessors() * 2 "

Reply:
Updated in